Libraries/WebSocket/RCTSRWebSocket.m

```diff
@@ -465,12 +472,21 @@ - (void)didConnect
   CFHTTPMessageSetHeaderFieldValue(request, CFSTR("Sec-WebSocket-Key"), (__bridge CFStringRef)_secKey);
   CFHTTPMessageSetHeaderFieldValue(request, CFSTR("Sec-WebSocket-Version"), (__bridge CFStringRef)[NSString stringWithFormat:@"%ld", (long)_webSocketVersion]);
 
-  CFHTTPMessageSetHeaderFieldValue(request, CFSTR("Origin"), (__bridge CFStringRef)_url.RCTSR_origin);
-
   if (_requestedProtocols) {
     CFHTTPMessageSetHeaderFieldValue(request, CFSTR("Sec-WebSocket-Protocol"), (__bridge CFStringRef)[_requestedProtocols componentsJoinedByString:@", "]);
   }
 
+  NSString *origin = (NSString *)[_requestedOptions objectForKey: @"origin"];
```

Comment:
Do more graceful type checking here -- `RCTLog` an error and fall back to the default origin if the value is not a string. Also prefer dictionary indexing syntax.